src/test/java/org/junit/jupiter/engine/AbstractJupiterTestEngineTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.jupiter.engine;
+
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+import static org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder.request;
+
+import org.junit.platform.engine.ExecutionRequest;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.UniqueId;
+import org.junit.platform.engine.test.event.ExecutionEventRecorder;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+
+/**
+ * Abstract base class for tests involving the {@link JupiterTestEngine}.
+ *
+ * @since 5.0
+ */
+public abstract class AbstractJupiterTestEngineTests {
+
+	private final JupiterTestEngine engine = new JupiterTestEngine();
+
+	protected ExecutionEventRecorder executeTestsForClass(Class<?> testClass) {
+		return executeTests(request().selectors(selectClass(testClass)).build());
+	}
+
+	protected ExecutionEventRecorder executeTests(LauncherDiscoveryRequest request) {
+		TestDescriptor testDescriptor = discoverTests(request);
+		ExecutionEventRecorder eventRecorder = new ExecutionEventRecorder();
+		engine.execute(new ExecutionRequest(testDescriptor, eventRecorder, request.getConfigurationParameters()));
+		return eventRecorder;
+	}
+
+	protected TestDescriptor discoverTests(LauncherDiscoveryRequest request) {
+		return engine.discover(request, UniqueId.forEngine(engine.getId()));
+	}
+
+}
```

Comment:
Have you considered checking the nullness of the parameters of all the methods in this class, using `Objects.requireNonNull` or Guava's `Preconditions.checkNotNull`?